diff --git a/include/faunus/analysis.h b/include/faunus/analysis.h
index 658fe74d..2f39ebe3 100755
--- a/include/faunus/analysis.h
+++ b/include/faunus/analysis.h
@@ -877,7 +877,7 @@ namespace Faunus {
                 d.cnt++;
                 d.tot = pot.g2g(spc.p, g1, g2); // exact el. energy
                 d.ii = a.charge * b.charge * rinv; // ion-ion, etc.
-                d.id = ( a.charge*b.mu.dot(r) - b.charge*a.mu.dot(r) ) * r3inv;
+                d.id = q2mu(a.charge*b.muscalar,b.mu,b.charge*a.muscalar,a.mu,r);
                 d.dd = mu2mu(a.mu, b.mu, a.muscalar*b.muscalar, r);
                 d.iq = q2quad(a.charge, b.theta, b.charge, a.theta, r);
 
